conda/gateways/download.py

```diff
@@ -73,7 +73,9 @@ def download(url, target_full_path, md5sum):
                 with open(target_full_path, 'wb') as fh:
                     streamed_bytes = 0
                     for chunk in resp.iter_content(2 ** 14):
-                        streamed_bytes += len(chunk)
+                        # chunk could be the decompressed form of the real data
+                        # but we want the exact number of bytes read till now
+                        streamed_bytes = resp.raw.tell()
```

Comment:
@kalefranz I kept it inside the for loop because we track the progress inside it (at line 87/89):

```bash
getLogger('fetch.update').info(streamed_bytes)
```